adoc/admin-monitoring-cluster-health-check.adoc

```diff
@@ -0,0 +1,372 @@
++
+The second method directly talks to kubelet can be used in more traditional monitoring where one must configure each node to be checked.
++
+** Configuration and Token retrieval
++
+Create a Service Account (monitoring) with a secondary Token (monitoring-secret-token) associated. The token will be used in HTTP requests to authenticate against the APIserver.
++
+This Service Account can only fetch information about nodes and pods. It is best practice to not use the default created token. Using a secondary token is also easier for management. Create a file [path]`kubelet.yaml` with the following as content.
++
+----
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: monitoring
+  namespace: kube-system
+secrets:
+- name: monitoring-secret-token
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: monitoring-secret-token
+  namespace: kube-system
+  annotations:
+    kubernetes.io/service-account.name: monitoring
+type: kubernetes.io/service-account-token
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: monitoring-clusterrole
+  namespace: kube-system
+rules:
+- apiGroups: [""]
+  resources:
+  - nodes/metrics
+  - nodes/proxy
+  - pods
+  verbs: ["get", "list"]
+- nonResourceURLs: ["/metrics", "/healthz", "/healthz/*"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: monitoring-clusterrole-binding
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: monitoring-clusterrole
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: monitoring
+  namespace: kube-system
+----
++
+Apply the yaml file
++
+[source,bash]
+----
+kubectl apply -f kubelet.yaml
+----
+Export the token to an environment variable:
++
+[source,bash]
+----
+TOKEN=$(kubectl -n kube-system get secrets monitoring-secret-token -o jsonpath='{.data.token}' | base64 -d)
+----
++
+This token can now be passed in headers in the form: "Authorization: Bearer $TOKEN"
++
+Now export important values as environment variables.
++
+
+** Testing Token Remotely
+. Choose a Kubernetes master node or worker node. The `NODE_IP` here must be a node's IP address or FQDN. The `NODE_NAME` here must be a node name in your Kubernetes cluster. Export the variables `NODE_IP` and `NODE_NAME` so it can be reused.
++
+[source,bash]
+----
+NODE_IP="10.86.4.158"
+NODE_NAME=worker0
+----
+
+. Retrieve the TOKEN with kubectl.
++
+[source,bash]
+----
+TOKEN=$(kubectl -n kube-system get secrets monitoring-secret-token -o jsonpath='{.data.token}' | base64 -d)
+----
+
+. Get APISERVER from the configuration file. You can skip this step if you only want to use the kubelet endpoint.
++
+[source,bash]
+----
+APISERVER=$(kubectl config view | grep server | cut -f 2- -d ":" | tr -d " ")
```

Comment:
I thin k it is good to mention that API Server is <LB_IP:6443>

Reply:
Change to use control plane IP/FQDN